testsuite/MDAnalysisTests/analysis/test_density.py

```diff
@@ -105,6 +112,67 @@ def test_check_set_unit_nolength(self, D):
         with pytest.raises(ValueError):
             D._check_set_unit(units)
 
+# 82 -> 84%; lines 757-758
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        assert D._check_set_unit(units) == None
+
+    # line 774
+    def test_check_set_unit_density_none(self, D):
+        units = {'density': None}
+        D._check_set_unit(units)
+        del D.units['density']
+        D._check_set_unit({})
+        assert D.units['density'] == None
+
+    # 84 -> 86%; lines 789-792
+    def test_parameters_isdensity(self, D):
+        with pytest.warns(UserWarning, match='Running make_density()'):
+            D.make_density()
+
+    # 86 -> 89%; lines 869-871
+    def test_check_convert_density_grid_not_density(self, D1):
+        unit = 'nm^{-3}'
+        with pytest.raises(RuntimeError, match="The grid is not a density"):
+            D1.convert_density(unit)
+
+    # 89 -> 92%; lines 874-880
+    def test_check_convert_density_value_error(self, D):
+        unit = 'A^{-2}'
+        with pytest.raises(ValueError, match="The name of the unit"):
+            D.convert_density(unit)
+
+    # 92 -> 93%; line 872
+    def test_check_convert_density_units_same_density_units(self, D):
+        unit = 'A^{-3}'
+        D.convert_density(unit)
+        assert unit == D.units['density']
+
+    # 93 -> 93%; line 881
+    def test_check_convert_density_units_density(self, D):
+        unit = 'nm^{-3}'
+        D.convert_density(unit)
+        assert D.units['density'] == unit
```

Comment:
Rather than just checking that the units have been set. It would be good to also check that numerically the conversion has been made in a way that would be expected.

Reply:
Note: it would be more than reasonable [if not even suggested] to have this test as part of the same test where you compare the grid value change (same for the convert_length stuff).

Reply:
Sure, I considered to move it to same function that check convert_density but was hesitating. Thanks.

Reply:
See my comment with `deepcopy` on how you could also check that the values changed:
```python
...
assert D.units['density'] ==  unit
assert_almost_equal(D.grid, 10**3 * D_orig.grid)   # or write 1e3 * ... but I wanted to make the factor clear
```